electrumsv/wallet.py

```diff
@@ -242,6 +241,41 @@ def __init__(self, wallet: 'Wallet', row: AccountRow,
         # TODO(rt12) BACKLOG formalise porting/storage of invoice data extracted from storage data
         self.invoices = InvoiceStore({})
 
+    @lru_cache(maxsize=2**16)
+    def script_bytes_to_scripthash(self, _script_bytes):
+        return sha256(_script_bytes)
+
+    @lru_cache(maxsize=2**16)
+    def scriptpubkey_to_bytes(self, script):
+        return bytes(script)
+
+    def scriptpubkey_to_scripthash(self, script):
+        _script_bytes = self.scriptpubkey_to_bytes(script)
+        return self.script_bytes_to_scripthash(_script_bytes)
+
+    def get_key_txos(self, tx_id: str, keyinstance_id: int) -> Optional[Set[int]]:
+        """get the set of all output indices in a given transaction for a given keyinstance id"""
+        script, _script_bytes, _object = self._get_cached_script(keyinstance_id)
+        if self._key_txos.get(tx_id) is not None:
+            try:
+                scripthash = self.script_bytes_to_scripthash(_script_bytes)
```

Comment:
Leading underscore indicates the type system should ignore, so script and script_bytes should switch use of underscore.